packages/design-system-react/src/components/icon/template.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Template for generating icon components.
+ * This template is used by SVGR to transform SVG files into React components.
+ * It provides consistent formatting and props handling for all icon components.
+ *
+ * @param {object} variables - The variables passed to the template
+ * @param {object} variables.imports - The component import statements
+ * @param {object} variables.interfaces - The component type interfaces
+ * @param {string} variables.componentName - The name of the component being generated
+ * @param {object} variables.props - The component props configuration
+ * @param {string} variables.jsx - The JSX code for the component
+ * @param {object} variables.exports - The component export statements
+ * @param {object} options - Template options
+ * @param {Function} options.tpl - Template literal tag function
+ * @returns {string} The formatted component code
+ */
+
+/**
+ * Generates a React component from SVG template variables
+ *
+ * @param {object} variables - SVGR template variables
+ * @param {object} options - Template options containing tpl function
+ * @param {Function} options.tpl - Template literal tag function
+ * @returns {string} Generated component code
+ */
+function template(variables, { tpl }) {
+  return tpl`
+import type { SVGProps, Ref } from 'react';
+import * as React from 'react';
+import { forwardRef } from 'react';
+
+const ${variables.componentName} = (
+  props: SVGProps<SVGSVGElement>,
+  ref: Ref<SVGSVGElement>,
+) => (
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 512 512"
+    fill="currentColor"
+    role="img"
+    ref={ref}
+    {...props}
+  >
+    ${variables.jsx}
+  </svg>
+);
+
+const ForwardRef = forwardRef(${variables.componentName});
+export default ForwardRef;
+`;
+}
+
+module.exports = template;
```

Comment:
The template.js file is a crucial part of our SVGR pipeline:
1. It's a custom template that controls how SVG files are transformed into React components
2. Must be JavaScript because SVGR's build process directly require()s and executes this file during transformation
3. The template:
- Sets up proper TypeScript types for the generated components
- Adds consistent props handling (SVGProps, ref forwarding)
- Ensures consistent SVG attributes (viewBox, xmlns, etc.)
- Maintains uniform component structure across all icons